src/main/java/com/laytonsmith/core/functions/PlayerManagement.java

```diff
@@ -3432,6 +3433,113 @@ public Construct exec(Target t, Environment environment, Construct... args) thro
 	}
 
 	@api(environments = {CommandHelperEnvironment.class})
+	public static class phas_storm extends AbstractFunction {
+
+		@Override
+		public ExceptionType[] thrown() {
+			return new ExceptionType[]{ExceptionType.PlayerOfflineException};
+		}
+
+		@Override
+		public boolean isRestricted() {
+			return true;
+		}
+
+		@Override
+		public Boolean runAsync() {
+			return false;
+		}
+
+		@Override
+		public Construct exec(Target t, Environment environment, Construct... args) throws ConfigRuntimeException {
+			MCPlayer m = environment.getEnv(CommandHelperEnvironment.class).GetPlayer();
+			if(args.length == 1) {
+				m = Static.GetPlayer(args[0], t);
+			}
+			return CBoolean.get(m.getPlayerWeather() == MCWeather.DOWNFALL);
+		}
+
+		@Override
+		public String getName() {
+			return "phas_storm";
+		}
+
+		@Override
+		public Integer[] numArgs() {
+			return new Integer[]{0, 1};
+		}
+
+		@Override
+		public String docs() {
+			return "void {[player]} Returns true if the given player is experiencing precipitation.";
```

Comment:
Does this return true if the global weather is raining, or just if set_pstorm(true) has been previously called? This should be documented here to avoid confusion.